.travis.yml

```diff
@@ -1,22 +1,34 @@
 # https://docs.haskellstack.org/en/stable/travis_ci/
-# run on containerized infrastructure hopefully
-dist: trusty
-sudo: false
+jobs:
+  include:
+    - os: linux
+      dist: xenial
+    - os: linux
+      dist: trusty
+    - os: linux
+      dist: bionic
+    - os: osx
+      osx_image: xcode11.6
+    - os: osx
+      osx_image: xcode11.3
+
 language: generic
 
 # Caching so the next build will be fast too.
 cache:
   directories:
   - $HOME/.stack
 
-before_cache:
 # Do any cleanup here to avoid caching junk.
+before_cache:
+  # this was getting corrupted in the cache for osx ??
+  - rm -rf /Users/travis/.stack/setup-exe-cache
```

Comment:
I have a support issue open with Travis for this.

And, I'd like to know if any other cache entries are getting corrupted, but I'm out of steam to:
1. delete the cache
2. construct the magic web request to launch a debug container
3. tmate in and manually run the build with no cache (half an hour)
4. figure out how to find + xargs + sha1sum
5. run regular build to populate cache (another half an hour)
6. run a debug build, tmate in, and re-sha1 the tree